File: tests/fixtures/sample.go
// Package sample provides sample types for testing.
package sample

import (
	"context"
	"fmt"
	"encoding/json"
)

// MaxRetries is the maximum number of retries.
const MaxRetries = 3

var counter int

// User represents a user in the system.
type User struct {
	ID    string `json:"id"`
	Name  string `json:"name"`
	Email string `json:"email"`
}

// Result is a generic result type.
type Result interface {
	Error() error
}

// Status represents the status of an entity.
type Status int

const (
	StatusActive   Status = iota
	StatusInactive
)

// UserService manages users.
type UserService struct {
	db interface{}
}

// NewUserService creates a new UserService.
func NewUserService(db interface{}) *UserService {
	return &UserService{db: db}
}

// FindUser finds a user by ID.
func (s *UserService) FindUser(ctx context.Context, id string) (*User, error) {
	return nil, nil
}

// Greet greets a person by name.
func Greet(name string) string {
	return fmt.Sprintf("Hello, %s!", name)
}

// FetchData fetches data from a URL.
func FetchData(url string) ([]byte, error) {
	data, _ := json.Marshal(url)
	return data, nil
}

File: tests/fixtures/sample.rs
use std::fmt;
use std::collections::HashMap;

/// Maximum retry count.
pub const MAX_RETRIES: u32 = 3;

static COUNTER: std::sync::atomic::AtomicU32 = std::sync::atomic::AtomicU32::new(0);

/// A user in the system.
#[derive(Debug, Clone)]
pub struct User {
    pub id: String,
    pub name: String,
    pub email: String,
}

/// Result type alias.
pub type Result<T> = std::result::Result<T, Box<dyn std::error::Error>>;

/// Status of an entity.
pub enum Status {
    Active,
    Inactive,
}

/// A trait for services.
pub trait Service {
    /// Initialize the service.
    fn init(&self) -> Result<()>;
}

impl User {
    /// Create a new user.
    pub fn new(id: String, name: String, email: String) -> Self {
        User { id, name, email }
    }
}

impl fmt::Display for User {
    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
        write!(f, "{} ({})", self.name, self.email)
    }
}

/// Greet a person by name.
pub fn greet(name: &str) -> String {
    format!("Hello, {}!", name)
}

/// Fetch data from a URL.
pub async fn fetch_data(url: &str) -> Result<Vec<u8>> {
    Ok(Vec::new())
}

fn private_helper() -> bool {
    true
}

mod internal {
    pub fn nested_public() {}
}
